src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```diff
@@ -0,0 +1,349 @@
+                    {
+                        string newPath = Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(path);
+
+                        // Unlike 'GetResolvedProviderPathFromPSPath'
+                        // 'GetUnresolvedProviderPathFromPSPath' does not check a file existence
+                        // so do that explicity
+                        if (File.Exists(newPath))
+                        {
+                            pathsToProcess.Add(newPath);
+                        }
+                        else
+                        {
+                            ItemNotFoundException pathNotFound =
+                                new ItemNotFoundException(
+                                    newPath,
+                                    "PathNotFound",
+                                    SessionStateStrings.PathNotFound);
+                            ErrorRecord errorRecord = new ErrorRecord(pathNotFound,
+                                "FileNotFound",
+                                ErrorCategory.ObjectNotFound,
+                                path);
+                            WriteError(errorRecord);
+                        }
+                    }
+                    break;
+            }
+
+            foreach (string path in pathsToProcess)
+            {
+                byte[] bytehash = null;
+                String hash = null;
+
+                try
+                {
+                    Stream openfilestream = File.OpenRead(path);
+                    bytehash = hasher.ComputeHash(openfilestream);
+
+                    hash = BitConverter.ToString(bytehash).Replace("-","");
+                    WriteHashResult(Algorithm, hash, path);
+                }
+                catch
+                {
+
+                }
+            }
+        }
+
+        /// <summary>
+        /// Create FileHashInfo object and output it
+        /// </summary>
+        private void WriteHashResult(String Algorithm, String hash, String path)
+        {
+            FileHashInfo result = new FileHashInfo();
+            result.Algorithm = Algorithm;
+            result.Hash = hash;
+            result.Path = path;
+            WriteObject(result);
+        }
+
+        /// <summary>
+        /// Parameter set names
+        /// </summary>
+        private const string PathParameterSet = "Path";
+        private const string LiteralPathParameterSet = "LiteralPath";
+        private const string StreamParameterSet = "StreamParameterSet";
+
+    }
+
+    /// <summary>
+    /// Base Cmdlet for cmdlets which deal with crypto hashes
+    /// </summary>
+    public class HashCmdletBase : PSCmdlet
+    {
+        /// <summary>
+        /// Algorithm parameter
+        /// The hash algorithm name: "SHA1", "SHA256", "SHA384", "SHA512", "MD5"
+        /// </summary>
+        /// <value></value>
+        [Parameter(Position = 1)]
+        [ValidateSet(HashAlgorithmNames.SHA1,
```

Comment:
I believe these are hashing algorithms, so security isn't the issue. My guess is less demand.

But the real question here - are we OK with a PR that would be a breaking change to Windows PowerShell if we took this code back to Windows as is?